core/transaction/transaction.go

```diff
@@ -339,6 +346,30 @@ func (tx *Transaction) GetProgramHashes() ([]Uint160, error) {
 			return nil, NewDetailErr(err, ErrNoCode, "[Transaction - BookKeeper], GetProgramHashes ToCodeHash failed.")
 		}
 		hashs = append(hashs, astHash)
+	case RegisterName:
+		registrant := tx.Payload.(*payload.RegisterName).Registrant
```

Comment:
If use programhash in payload, these code could be moved to transaction construction layer.

Reply:
Need to do transform from encoded public key to program hash. Maybe need a function like following:
```
func CreateSignatureRedeemScriptWithEncodedPublicKey(encodedPublicKey []byte) ([]byte, error) {
	sb := pg.NewProgramBuilder()
	sb.PushData(encodedPublicKey)
	sb.AddOp(vm.CHECKSIG)
	return sb.ToArray(), nil
}
```

Reply:
Done. Can resolve?